wgpu-core/src/logging/subscriber.rs

```diff
@@ -0,0 +1,212 @@
+use std::{
+    borrow::Cow,
+    fmt, io,
+    io::Write,
+    path::Path,
+    sync::{Arc, Mutex},
+    time::Instant,
+};
+use tracing::{
+    field::{Field, Visit},
+    span, Event, Metadata, Subscriber,
+};
+use tracing_subscriber::{
+    layer::{Context, SubscriberExt},
+    registry::LookupSpan,
+    EnvFilter, Layer,
+};
+
+/// Set up the "standard" logger.
+///
+/// This is fairly inflexible, but a good default to start with. If you need more customization,
+/// take what this function does and implement it however you need.
+///
+/// # Args
+///
+/// - `chrome_tracing_path` if set to `Some`, will create a trace compatible with chrome://tracing
+///   at that location.  
+pub fn initialize_default_subscriber(chrome_trace_path: Option<impl AsRef<Path>>) {
+    let chrome_tracing_layer_opt =
+        chrome_trace_path.map(|path| ChromeTracingLayer::with_file(path).unwrap());
+
+    // Tracing currently doesn't support type erasure with layer composition
+    if let Some(chrome_tracing_layer) = chrome_tracing_layer_opt {
+        tracing::subscriber::set_global_default(
+            tracing_subscriber::Registry::default()
+                .with(chrome_tracing_layer)
```

Comment:
if this is the only line that's different, can we move out the whole thing from the if/else arms?

Reply:
They have different types, `.with` results in `Layered<TheLastThing, ThisThing>` so I have to carry it through all the way to the point of type erasure (set_global_default)